move Tie::Hash::NamedCapture into the core executable

tonycoz opened the pull request:
This improves compatibility with minitest

tonycoz committed:
move the implementation of %-, %+ into core

Previousl this could cause problems during minitest.

Fixes #17293

tonycoz committed:
POSIX isn't loadable in miniperl

jkeenan commented:
I've made this available for smoke testing in the smoke-me/jkeenan/tonycoz-17293-named-capture-miniperl branch.

This addresses the problem with t/re/pat.t during `make minitest` discussed in https://github.com/Perl/perl5/issues/17293.  But it does so by skipping that file entirely `if_miniperl`.  Is that intended?

Thank you very much.
Jim Keenan

jkeenan commented:
> I've made this available for smoke testing in the smoke-me/jkeenan/tonycoz-17293-named-capture-miniperl branch.
> 
> This addresses the problem with t/re/pat.t during `make minitest` discussed in #17293. But it does so by skipping that file entirely `if_miniperl`. Is that intended?
> 

More specifically, are there test blocks within t/re/pat.t which are **not** POSIX-dependent and which should **still** be tested during `make minitest`?

If so, then we should probably place those blocks in a file of their own.

jkeenan commented:
> > I've made this available for smoke testing in the smoke-me/jkeenan/tonycoz-17293-named-capture-miniperl branch.
> > This addresses the problem with t/re/pat.t during `make minitest` discussed in #17293. But it does so by skipping that file entirely `if_miniperl`. Is that intended?
> 
> More specifically, are there test blocks within t/re/pat.t which are **not** POSIX-dependent and which should **still** be tested during `make minitest`?
> 
> If so, then we should probably place those blocks in a file of their own.

Yes, we can move some tests out of t/re/pat.t to a new file that will run under `make minitest`.  See: https://github.com/tonycoz/perl5/pull/1

Thank you very much.
Jim Keenan

tonycoz committed:
fix skip for loop over locales

Any skips inside this loop was skipping over the entire loop,
not just the current iteration, from the conditions tested in
the loop this seemed incorrect (besides messing up the test count.)

tonycoz committed:
skip code that requires dynamic loading and minitest works

hvds commented on vxs.inc:
```diff
@@ -5,7 +5,7 @@
 #  define VXSp(name) XS_##name
 /* VXSXSDP = XSUB Details Proto */
-#  define VXSXSDP(x) x
+#  define VXSXSDP(x) x, 0
```
This looks problematic given the file's header. I think it expects to retain back compat, which may be hard with the current approach.

tonycoz replied:
This is only used when version is part of core, it can only be problem if we backport version to maint-older, which we don't typically.

demerphq commented:
+1 on the idea. I havent  reviewed the implementation. FWIW, I think this
brings the implementation full circle. I originally implemented this with a
set of utility functions in universal.c and then the rest was Perl that
called into it (hence we never had this problem when the code was
originally implemented) and the implementation was later redone as XS in
its own file.

Yves

On Sun, 17 Nov 2019 at 22:32, Tony Cook <notifications@github.com> wrote:

> This improves compatibility with minitest
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Perl/perl5/pull/17305
> Commit Summary
>
>    - move the implementation of %-, %+ into core
>    - POSIX isn't loadable in miniperl
>
> File Changes
>
>    - *M* MANIFEST <https://github.com/Perl/perl5/pull/17305/files#diff-0>
>    (1)
>    - *M* ext/Tie-Hash-NamedCapture/NamedCapture.pm
>    <https://github.com/Perl/perl5/pull/17305/files#diff-1> (8)
>    - *D* ext/Tie-Hash-NamedCapture/NamedCapture.xs
>    <https://github.com/Perl/perl5/pull/17305/files#diff-2> (134)
>    - *M* t/op/magic.t
>    <https://github.com/Perl/perl5/pull/17305/files#diff-3> (10)
>    - *M* t/re/pat.t
>    <https://github.com/Perl/perl5/pull/17305/files#diff-4> (4)
>    - *M* universal.c
>    <https://github.com/Perl/perl5/pull/17305/files#diff-5> (214)
>    - *M* vxs.inc <https://github.com/Perl/perl5/pull/17305/files#diff-6>
>    (2)
>
> Patch Links:
>
>    - https://github.com/Perl/perl5/pull/17305.patch
>    - https://github.com/Perl/perl5/pull/17305.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Perl/perl5/pull/17305?email_source=notifications&email_token=AAAZ5R35RRC27TQ3ZS2ZPOTQUGZ6VA5CNFSM4JOL33U2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HZ4MIYA>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZ5RZ7WY6V4QJNXNUMK3LQUGZ6VANCNFSM4JOL33UQ>
> .
>


-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

tonycoz commented:
> This addresses the problem with t/re/pat.t during `make minitest` discussed in #17293. But it does so by skipping that file entirely `if_miniperl`. Is that intended?

I'm confused by this comment, one of the commits removes:
````
skip_all('no re module') unless defined &DynaLoader::boot_DynaLoader;
````
allowing the code to run under miniperl.

skips are added to some other tests, eg. skipping a test that requires Encode, or POSIX.

If you have a specific line in mind here, you can select code on the right side of the commit displays and comment on the specific code in question.

jkeenan commented:
> > This addresses the problem with t/re/pat.t during `make minitest` discussed in #17293. But it does so by skipping that file entirely `if_miniperl`. Is that intended?
> 
> I'm confused by this comment, one of the commits removes:
> 
> ```
> skip_all('no re module') unless defined &DynaLoader::boot_DynaLoader;
> ```
> 
> allowing the code to run under miniperl.
> 
> skips are added to some other tests, eg. skipping a test that requires Encode, or POSIX.
> 
> If you have a specific line in mind here, you can select code on the right side of the commit displays and comment on the specific code in question.

I suspect I was probably referring to t/re/pat_advanced.t rather than t/re/pat.t.  Investigating.

Thank you very much.
Jim Keenan

jkeenan commented:
> > > This addresses the problem with t/re/pat.t during `make minitest` discussed in #17293. But it does so by skipping that file entirely `if_miniperl`. Is that intended?
> > 
> > 
> > I'm confused by this comment, one of the commits removes:
> > ```
> > skip_all('no re module') unless defined &DynaLoader::boot_DynaLoader;
> > ```
> > 
> > 
> > allowing the code to run under miniperl.
> > skips are added to some other tests, eg. skipping a test that requires Encode, or POSIX.
> > If you have a specific line in mind here, you can select code on the right side of the commit displays and comment on the specific code in question.
> 
> I suspect I was probably referring to t/re/pat_advanced.t rather than t/re/pat.t. Investigating.
> 

It turns out that the problem was that I incorrectly created a local branch from your pull request.  In that malformed branch `t/re/pat.t` was being skipped in its entirety.

I have just now created a new branch (`smoke-me/jkeenan/tonycoz-17293-named-capture-miniperl`) from your pull request.  When I configure this branch and run `make minitest`, `t/re/pat.` is no longer skipped.  Here are the tests that are run in that branch under `make minitest`:
```
t/re/pat ....................... ok
t/re/pat_advanced .............. skipped
t/re/pat_advanced_thr .......... skipped
t/re/pat_psycho ................ ok
t/re/pat_psycho_thr ............ skipped
t/re/pat_re_eval ............... ok
t/re/pat_re_eval_thr ........... skipped
t/re/pat_rt_report ............. skipped
t/re/pat_rt_report_thr ......... skipped
t/re/pat_special_cc ............ ok
t/re/pat_special_cc_thr ........ skipped
t/re/pat_thr ................... skipped
```
I apologize for the confusion.

Thank you very much.
Jim Keenan

jkeenan commented:
+1 to merge.
